### PR TITLE
wiz-broker: Take TLS verification skip and CA certificate out of HTTP proxy configuration

### PR DESCRIPTION
**Pull request by nivbend**
General idea here is that neither skipping TLS or the certificate are related to the HTTP proxy and shouldn't be coupled with it - So I took them out of that secret and removed the dependency on whether HTTP proxy configuration is enabled to set the environment variable (for skip TLS) or mounting the certificate.

### CODE REVIEW

File: wiz-broker/templates/secrets.yaml [hunk 1]
```diff
@@ -15,6 +15,23 @@ type: Opaque
...
+  name: {{ include "wiz-broker.caCertificateSecretName" . | trim }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    {{- include "wiz-broker.labels" . | nindent 4 }}
+  annotations:
+    {{- with .Values.global.broker.caCertificate.annotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+data:
+  caCertificate: {{ .Values.global.broker.caCertificate.pem | b64enc | quote }}
```
Comment (eyal-moscovici) [inline wiz-broker/templates/secrets.yaml:33]: maybe use stringData without  b64enc

File: wiz-broker/templates/wiz-broker-deployment.yaml [hunk 2]
```diff
@@ -1,10 +1,9 @@
 {{- if .Values.global.broker.enabled }}
 {{ $mountPath := "/etc/connectorData" }}
-{{ $sslMountPath := "/etc/ssl/certs" }}
 {{ $connectorDataFileName := "data" }}
 {{ $connectorDataFilePath := printf "%s/%s" $mountPath "data" }}
+{{ $sslMountPath := "/etc/ssl/certs" }}
 {{ $customCertificateFileName := "ca-certificates.crt" }}
-{{ $customCertificateFilePath := printf "%s/%s" $sslMountPath $customCertificateFileName }}
 apiVersion: apps/v1
```
Comment (eyal-moscovici) [inline wiz-broker/templates/wiz-broker-deployment.yaml:7]: Why did you remove this?

File: wiz-broker/templates/wiz-broker-deployment.yaml [hunk 3]
```diff
@@ -125,16 +122,14 @@ spec:
...
           - name: DISABLE_TLS_VALIDATION
-            valueFrom:
-              secretKeyRef:
-                name: {{ include "wiz-broker.proxySecretName" . | trim }}
-                key: skipTlsVerify
-                optional: false
-          {{- if .Values.global.httpProxyConfiguration.caCertificate }}
-          - name: CA_CERT_PATH
-            value: {{ $customCertificateFilePath }}
+            value: true
```
Comment (eyal-moscovici) [inline wiz-broker/templates/wiz-broker-deployment.yaml:128]: Take the value maybe instead of setting to true?

File: wiz-broker/templates/wiz-broker-deployment.yaml (hunk 2 above)
Comment (nivbend) [reply wiz-broker/templates/wiz-broker-deployment.yaml:7]: Moved to where it's used.

File: wiz-broker/templates/wiz-broker-deployment.yaml (hunk 3 above)
Comment (nivbend) [reply wiz-broker/templates/wiz-broker-deployment.yaml:128]: We only define this if `skipTlsVerify` is true, otherwise I won't define the environment variable.

File: wiz-broker/templates/secrets.yaml (hunk 1 above)
Comment (nivbend) [reply wiz-broker/templates/secrets.yaml:33]: Tried playing around with it, but it'd be harder to get newlines right and might fail the yaml parsing. PEM files probably contain newlines...

**Review by eyal-moscovici — APPROVED** [review]
